lib/que/adapters/yugabyte.rb

```diff
@@ -0,0 +1,75 @@
+# frozen_string_literal: true
+
+# https://github.com/que-rb/que/blob/80d6067861a41766c3adb7e29b230ce93d94c8a4/lib/que/active_job/extensions.rb
+module Que
+    module Adapters
+      class Yugabyte < Que::Adapters::ActiveRecord
+        def initialize
+            super
+        end
+
+        def checkout_activerecord_adapter(&block)
+            YugabyteRecord.connection_pool.with_connection(&block)
```

Comment:
we need to open seperate connection for each worker thread and pass it. In which case it might work. 
Adapter is initialised only once. I think this is to make sure each worker thread uses same connection to execute the job and to acquire and release the lock